packages/paste-core/components/button/src/styles.ts

```diff
@@ -9,70 +9,75 @@ import {ButtonWrapperProps, ButtonChildrenProps} from './types';
  * Sizes
  */
 const sizeReset = css`
+  font-size: 100%;
   padding: 0;
 `;
 const sizeIcon = (props: ButtonWrapperProps): SerializedStyles => css`
   padding: ${themeGet('space.space30')(props)};
   border-radius: ${themeGet('radii.borderRadius20')(props)};
+  font-size: 100%;
   /* To fix abnormal button padding-bottom */
   line-height: unset;
 `;
 const sizeSmall = (props: ButtonWrapperProps): SerializedStyles => css`
   padding: ${themeGet('space.space10')(props)} ${themeGet('space.space30')(props)};
   border-radius: ${themeGet('radii.borderRadius10')(props)};
   font-size: ${themeGet('fontSizes.fontSize30')(props)};
-  line-height: 24px;
+  line-height: ${themeGet('lineHeights.lineHeight30')(props)};
 `;
 const sizeDefault = (props: ButtonWrapperProps): SerializedStyles => css`
   padding: ${themeGet('space.space30')(props)} ${themeGet('space.space60')(props)};
   border-radius: ${themeGet('radii.borderRadius20')(props)};
   font-size: ${themeGet('fontSizes.fontSize30')(props)};
   font-weight: ${themeGet('fontWeights.fontWeightSemibold')(props)};
-  line-height: 24px;
+  line-height: ${themeGet('lineHeights.lineHeight30')(props)};
 `;
 
 /*
  * Base
  */
-const baseButtonWrapper = css`
+const baseButtonWrapper = (props: ButtonWrapperProps): SerializedStyles => css`
   /* Hide default browser styles */
   appearance: none;
   border: none;
   display: inline-block;
   outline: none;
   background: none;
-  font-size: 100%;
```

Comment:
Moving this to where it's needed. The ordering was weird so even though this is added first for each button, 100% was coming last and overriding the font sizes